net.go

```diff
@@ -1088,6 +1123,8 @@ func (m *Memberlist) decryptRemoteState(bufConn io.Reader, streamLabel string) (
 	// Ensure we aren't asked to download too much. This is to guard against
 	// an attack vector where a huge amount of state is sent
 	moreBytes := binary.BigEndian.Uint32(cipherText.Bytes()[1:5])
+	metrics.SetGaugeWithLabels([]string{"memberlist", "size", "remote"}, float32(moreBytes), m.metricLabels)
```

Comment:
Is this actually helpful in a gauge?  Should I remove this?  I added this here because this is where warnings and errors happen related to maxPushStateBytes.  The receiver warns that the incoming message is too large.  
Since it appears that Gauges get reported with the hostname embedded in the metric, IE `consul_server_0_memberlist_size_remote`, this metric would seem to indicate "What is the size of the last message that Server 0 received?"  if we wanted to see it broken out by sender, we could add a label, but this seems like it is not that effective and the local state is really a much better thing to follow.  Thoughts?

Reply:
Yeah it seems like if you keep this one (remote), it would make more sense as like a histogram so you could avoid collapsing the data points in the manner of a gauge.

Reply:
👍 thanks @rboyer. I've made the suggested change.